pkgs/desktops/mate/atril/default.nix

```diff
@@ -26,6 +47,18 @@ stdenv.mkDerivation rec {
     mate.mate-desktop
     hicolor-icon-theme
     texlive.bin.core  # for synctex, used by the pdf back-end
+    (optionals (enableDjvu) djvulibre)
+    (optionals (enableEpub) webkitgtk)
+    (optionals (enablePostScript) libspectre)
+    (optionals (enableXps) libgxps)
+  ];
+
```

Comment:
`optionals` always returns a list. When its condition is false, the list is empty. Otherwise it has one element. Therefore it is not approppriate inside the list of build dependences. Instead the result of `optionals` should be appended to a list with the other elements.

Something like this:

``` nix
  buildInputs = [
    gtk3
    glib
    itstool
    libsecret
    libxml2
    poppler
    mate.caja
    mate.mate-desktop
    hicolor-icon-theme
    texlive.bin.core  # for synctex, used by the pdf back-end
  ]
  ++ optionals enableDjvu djvulibre
  ++ optionals enableEpub webkitgtk
  ++ optionals enablePostScript libspectre
  ++ optionals enableXps libgxps
  ;
```

`optionals` is documented at https://nixos.org/nixpkgs/manual/#sec-functions-library